### ISSUE TITLE
Cannot find docker after running InstallerTask

### ISSUE THREAD

**Issue by Gamecock** [microsoft/azure-pipelines-tasks#15133]
**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: DockerInstaller 

## Environment
- Server - Azure Pipelines    
    
    - If using Azure Pipelines, provide the account name, team project name, build definition name/build number: 
      account: jblprd
      project: JAMN Project
      pipeline: Autonestserver

- Agent - Hosted 
    Windows Server 2019
    Container agent per [docs](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/docker?view=azure-devops#windows)

## Issue Description
Run a Job With Tasks DockerInstaller,  and Task Docker.
Docker Installer Succeeds:
Docker fails with 
`##[error]Unhandled: Unable to locate executable file: 'docker'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also verify the file has a valid extension for an executable file.`

### Task logs

[Enable debug logging and please provide the zip file containing all the logs for a speedy resolution]

## Troubleshooting
Checkout how to troubleshoot failures and collect debug logs: [logs_398793.zip](https://github.com/microsoft/azure-pipelines-tasks/files/6956817/logs_398793.zip)

### Error logs
Docker Installer Task
```
Starting: Docker Installer
==============================================================================
Task         : Docker CLI installer
Description  : Install Docker CLI on agent machine.
Version      : 0.181.0
Author       : Microsoft Corporation
Help         : https://aka.ms/azpipes-docker-installer-tsg
==============================================================================
Found tool in cache: docker-stable 20.10.8 x64
Prepending PATH environment variable with directory: C:\azp\agent\_work\_tool\docker-stable\20.10.8\x64
Verifying docker installation...
C:\azp\agent\_work\_tool\docker-stable\20.10.8\x64\docker.exe --version
Docker version 20.10.8, build 3967b7d
Finishing: Docker Installer
```

Docker Task
```
Starting: Build image
==============================================================================
Task         : Docker
Description  : Build or push Docker images, login or logout, start or stop containers, or run a Docker command
Version      : 2.188.0
Author       : Microsoft Corporation
Help         : https://aka.ms/azpipes-docker-tsg
==============================================================================
##[error]Unhandled: Unable to locate executable file: 'docker'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also verify the file has a valid extension for an executable file.
Finishing: Build image
```

**Comment by Gamecock** [comment]
[logs_398793.zip](https://github.com/microsoft/azure-pipelines-tasks/files/6956819/logs_398793.zip)
Logs from run

**Comment by Gamecock** [comment]
[logs_398804.zip](https://github.com/microsoft/azure-pipelines-tasks/files/6956958/logs_398804.zip)
Logs with system diagnostics enabled.

**Comment by JuanCSoto** [comment]
I have the same problem.

**Comment by celluj34** [comment]
I am having this same issue too - I found [this issue](https://github.com/microsoft/azure-pipelines-tasks/issues/12823) which describes a similar problem, but was closed. I think that was incorrect, since in the output of the job it says `Prepending PATH environment variable with directory`, which means it should be available for the duration of the job.

The title of the task is _Docker Installer_, but doesn't seem to actually _install_ anything. Why do I need to install docker onto the build machine itself when I should be able to use this task?
